src/lib.rs

```diff
@@ -49,60 +49,56 @@ cfg_if! {
     use source::Loader;
     use source::Reporter;
 
-    /// Create a module graph, based on loading and recursively analyzing the
-    /// dependencies of the module, returning the resulting graph.
-    #[allow(clippy::too_many_arguments)]
-    pub async fn create_graph(
+    #[derive(Default)]
+    pub struct CreateGraphOptions<'a> {
+      pub is_dynamic: bool,
+      pub maybe_imports: Option<Vec<(ModuleSpecifier, Vec<String>)>>,
+      pub maybe_resolver: Option<&'a dyn Resolver>,
+      pub maybe_locker: Option<Rc<RefCell<Box<dyn Locker>>>>,
+      pub maybe_parser: Option<&'a dyn SourceParser>,
+      pub maybe_reporter: Option<&'a dyn Reporter>,
+    }
+
+    async fn create_graph_inner(
```

Comment:
This is confusing and feels unnecessary, if we are refactoring, let's refactor.

Reply:
I assume the suggestion is to just have one `create_graph()` with the `BuildKind` in options since we're breaking anyway, or at least I think we should do that.

Reply:
Yes, as well as refactor the builder to use an "options bag" as well.  Might as well go for it.